File: TournamentManager.Contracts/Dtos/MatchDto.cs

using TournamentManager.Domain.Enums;

namespace TournamentManager.Contracts.Dtos
{
    public record MatchDto(int Id, int? HomeTeamId, int? AwayTeamId, string? HomeTeamName, string? AwayTeamName, int? HomeTeamScore, int? AwayTeamScore, DateTime? StartDate, Round Round);
}

File: TournamentManager.Frontend/Client/TournamentManager/src/models/matchDto.ts
import { Round } from '../enums/round'

export interface MatchDto {
    id: number | undefined
    homeTeamId: number | undefined
    awayTeamId: number | undefined
    homeTeamName: string | undefined
    awayTeamName: string | undefined
    homeTeamScore: number | undefined
    awayTeamScore: number | undefined
    startDate: string
    round: Round
}
